developer-docs/upgrading_kubernetes.md

```diff
@@ -6,17 +6,88 @@ From time to time we need to update the version of Kubernetes used by RKE2. This
 
 ### Container Image
 
-Create a new release tag at https://github.com/rancher/image-build-kube-proxy
+Create a new release tag at the [image-build-kube-proxy](https://github.com/rancher/image-build-kube-proxy) repo.
+
+* Click "Releases"
+* Click "Draft a new release"
+* Enter the new release number (the new k8s version) into the "Tag version" box
+* Click the "Publish release" button. 
+
+This will take a few minutes for CI to run but upon completion, a new image will be available in [Dockerhub](https://hub.docker.com/r/rancher/hardened-kube-proxy).
 
 ### Helm Chart
 
-Create a new release asset for in the [rke2-charts](github.com/rancher/rke2-charts) repository. Instructions for doing so can be found in the repo. This is necessary as the RKE2 build process will check for that chart and source it into one of its build artifacts.
+RKE2 depends on it's [Helm Charts](https://github.com/rancher/rke2-charts) being up-to-date with the expected versions for the Kubernetes components. The build process downloads these charts and bundles them into the runtime image.
+
+Create a PR in [rke2-charts](https://github.com/rancher/rke2-charts) that increments the version of `kube-proxy` in `packages/rke2-kube-proxy/charts/values.yaml`. Upon getting 2 approvals and merging, CI will create the needed build artifact that RKE2 will use.
```

Comment:
I am confused about this part, by the comment the version key [prints the version information and quit](https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-kube-proxy/charts/values.yaml#L215-L216). Do you mean the `image.tag`?

Reply:
@MonzElmasry Can you speak to this question?

Reply:
It looks like someone just took the command-line help from the kube-proxy command and applied it to each of the values.yaml fields, which is a bit confusing.

When @briandowns said 'increment the version' he didn't mean change the version field, he did (as you deduced) mean to increment the version of the image tag in `image.tag`, and also to increment the versions in Chart.yaml

This could probably be spelled out a little more plainly though:
```suggestion
Create a PR in [rke2-charts](https://github.com/rancher/rke2-charts) that updates the version of the `kube-proxy` image in both the `image.tag` field of `packages/rke2-kube-proxy/charts/values.yaml`, and also in `packages/rke2-kube-proxy/charts/Chart.yaml`. Upon getting 2 approvals and merging, CI will create the needed build artifact that RKE2 will use.
```